### PR TITLE
makes all food forkable

### PR DESCRIPTION
**Pull request by Pontenerd**
## About The Pull Request

Lets you eat all food with forks (minus soups!)

## Why It's Good For The Game

Forks are good for eating

## Testing Photographs and Procedure
https://cdn.discordapp.com/attachments/602204705862582341/1044003786575855616/2022-11-20_21-37-43.mp4

## Changelog
:cl:
tweak: tweaked fork code to allow you to eat all foods
/:cl:

### CODE REVIEW

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]
Please port this to Beestation as well, or move the changes to the nsv13 subfolder if they do not want this change.

**Comment by Bokkiewokkie** [comment]
Untestmerging as this is breaking other attackby() interactions with food!

File: _maps/map_files/Atlas/atlas2.dmm [hunk 1]
```diff
@@ -10197,9 +10197,6 @@
 /obj/machinery/atmospherics/pipe/simple/supply/hidden/layer2{
 	dir = 4
 	},
-/obj/machinery/light{
-	dir = 1
-	},
```
Comment (Bokkiewokkie) [inline _maps/map_files/Atlas/atlas2.dmm:10202]: This is causing a conflict

File: code/modules/food_and_drinks/food/snacks.dm [hunk 2]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+
+				var/datum/reagent/R = pick(reagents.reagent_list)
+				reagents.remove_reagent(R.type, 1)
+				F.forkload = R
+				if(reagents.total_volume <= 0)
+					qdel(src)
+			return ..()
+		else
+			to_chat(user, "<span class='warning'>You can't eat this with a spoon!</span>")
+
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:430]:
```suggestion
			return
	..()
```
You should probably let every other thing go about the normal interaction :p

File: code/modules/food_and_drinks/food/snacks.dm [hunk 3]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+				F.icon_state = "spoonloaded"
+				user.visible_message("[user] takes a spoonful of [src] with [user.p_their()] spoon!", \
+					"<span class='notice'>You take a spoonful of [src] with your spoon.</span>")
+
+				var/datum/reagent/R = pick(reagents.reagent_list)
+				reagents.remove_reagent(R.type, 1)
+				F.forkload = R
+				if(reagents.total_volume <= 0)
+					qdel(src)
+			return ..()
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:427]:
```suggestion
			return
```

File: code/modules/food_and_drinks/food/snacks.dm [hunk 4]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+		if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+			M = user
+		return eyestab(M,user)
+	else
+		return ..()
+
+
+/obj/item/kitchen/spoon/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(!istype(M))
+		return ..()
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:456]:
```suggestion
/obj/item/kitchen/spoon/attack(mob/living/carbon/M, mob/living/carbon/user)
	. = ..()
	if(!istype(M))
		return
```

File: code/modules/food_and_drinks/food/snacks.dm [hunk 5]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+			M.reagents.add_reagent(forkload.type, 1)
+		icon_state = "spoon"
+		forkload = null
+
+	else if(user.zone_selected == BODY_ZONE_PRECISE_EYES)
+		if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+			M = user
+		return eyestab(M,user)
+	else
+		return ..()
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:473]:
```suggestion
		return
```

File: code/modules/food_and_drinks/food/snacks.dm [hunk 6]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+				if(reagents.total_volume <= 0)
+					qdel(src)
+			return ..()
+		else
+			to_chat(user, "<span class='warning'>You can't eat this with a spoon!</span>")
+
+
+/obj/item/kitchen/fork/attack(mob/living/carbon/M, mob/living/carbon/user)
+	if(!istype(M))
+		return ..()
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:434]:
```suggestion
/obj/item/kitchen/fork/attack(mob/living/carbon/M, mob/living/carbon/user)
	. = ..()
	if(!istype(M))
		return
```

File: code/modules/food_and_drinks/food/snacks.dm [hunk 7]
```diff
@@ -390,3 +390,84 @@ All foods are distributed among various categories. Use common sense.
...
+		icon_state = "fork"
+		forkload = null
+
+	else if(user.zone_selected == BODY_ZONE_PRECISE_EYES)
+		if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+			M = user
+		return eyestab(M,user)
+	else
+		return ..()
+
```
Comment (Bokkiewokkie) [inline code/modules/food_and_drinks/food/snacks.dm:452]:
```suggestion
		return
```

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

**Review by Bokkiewokkie — DISMISSED** [review]

**Review by covertcorvid — DISMISSED** [review]

**Review by covertcorvid — APPROVED** [review]

**Comment by Bokkiewokkie** [comment]
Please update the [Beestation PR](https://github.com/BeeStation/BeeStation-Hornet/pull/8202) as wel.

**Pull request merged**